crates/typst/src/layout/repeat.rs

```diff
@@ -76,15 +79,27 @@ fn layout_repeat(
         frame.set_baseline(piece.baseline());
     }
 
+    let mut gap = elem.gap(styles).resolve(styles);
+    let fill = regions.size.x;
+    let width = piece.width();
+    let count = ((fill + gap) / (width + gap)).floor();
+    let remaining = (fill + gap) % (width + gap);
```

Comment:
I think the `fill + gap` calculation here is a bit hard to follow intuitively (at least, it's not directly obvious to me why it is correct). Could you add a brief comment explaining it?